src/WingetCreateCLI/UserSettings.cs

```diff
@@ -0,0 +1,170 @@
+        /// <param name="path">Output path of settings json file.</param>
+        public static void GenerateFileFromLoadedSettings(string path)
+        {
+            string output = JsonConvert.SerializeObject(Settings, Formatting.Indented);
+            File.WriteAllText(path, output);
+        }
+
+        /// <summary>
+        /// Loads a json file from a path and creates a SettingsManifest object model.
+        /// </summary>
+        /// <param name="path">Path to settings json file.</param>
+        /// <returns>SettingsManifest object model.</returns>
+        public static SettingsManifest LoadJsonFile(string path)
+        {
+            using (StreamReader r = new StreamReader(path))
+            {
+                string json = r.ReadToEnd();
+                return JsonConvert.DeserializeObject<SettingsManifest>(json);
+            }
+        }
+
+        /// <summary>
+        /// Checks if the tool is being launched for the first time.
+        /// </summary>
+        public static void FirstRunTelemetryConsent()
+        {
+            if (!File.Exists(SettingsJsonPath))
+            {
+                Prompt.Symbols.Done = new Symbol(string.Empty, string.Empty);
+                Prompt.Symbols.Prompt = new Symbol(string.Empty, string.Empty);
+                Console.WriteLine(Resources.TelemetrySettings_Message);
+                Console.WriteLine("------------------");
+                Console.WriteLine(Resources.TelemetryJustification_Message);
+                Console.WriteLine(Resources.TelemetryAnonymous_Message);
+                TelemetryDisable = !Prompt.Confirm(Resources.EnableTelemetryFirstRun_Message);
+            }
+        }
+
+        /// <summary>
+        /// Loads the correct settings file based on the following order.
+        /// 1. If the settings file exists and is valid, then overwrite the backup file with the current settings file.
+        /// 2. If the backup settings file exists and is valid, then recreate the settings file using the backup.
+        /// 3. Otherwise, use default settings based on the settings schema.
+        /// </summary>
+        private static void LoadSettings()
+        {
+            if (File.Exists(SettingsJsonPath) && ParseJsonFile(SettingsJsonPath).IsValid)
+            {
+                Settings = LoadJsonFile(SettingsJsonPath);
+            }
+            else if (File.Exists(SettingsBackupJsonPath) && ParseJsonFile(SettingsBackupJsonPath).IsValid)
+            {
+                Logger.WarnLocalized(nameof(Resources.UnexpectedErrorLoadSettings_Message));
+                Logger.WarnLocalized(nameof(Resources.LoadSettingsFromBackup_Message));
+                Settings = LoadJsonFile(SettingsBackupJsonPath);
+            }
+            else
+            {
+                // If either of these files exist, then this is not the first run and we can display warnings.
+                if (File.Exists(SettingsJsonPath) || File.Exists(SettingsBackupJsonPath))
```

Comment:
Comment doesn't seem to quite match scenario. I'd mention that we're here not just because the files exist, but because parsing them faile

Reply:
Fixed comment to more accurately reflect that these warnings are due to parsing errors